gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/MysqlMultiActiveLeaseArbiter.java

```diff
@@ -122,7 +122,8 @@ protected interface CheckedFunction<T, R> {
   // Need to define three separate statements to handle cases where row does not exist or has null values to check
   protected static final String CONDITIONALLY_ACQUIRE_LEASE_IF_NEW_ROW_STATEMENT = "INSERT INTO %s (flow_group, "
       + "flow_name, flow_execution_id, flow_action, event_timestamp, lease_acquisition_timestamp) "
-      + "VALUES (?, ?, ?, ?, CURRENT_TIMESTAMP, CURRENT_TIMESTAMP)";
+      + "SELECT ?, ?, ?, ?, CURRENT_TIMESTAMP, CURRENT_TIMESTAMP WHERE NOT EXISTS (SELECT * FROM %s "
+      + WHERE_CLAUSE_TO_MATCH_KEY + ")";
```

Comment:
rather than the `NOT EXISTS` can we simply rely on the uniqueness constraint for the composite PK `(fg, fn, fexid, flow_action)`?

Reply:
I am not sure if our DB settings for MySQL will throw an error if insert fails or UPSERT by default or ignore the insert if PK already exists so i wanted to use a statement that I can predict the behavior from

Reply:
`If you use the IGNORE modifier, ignorable errors that occur while executing the [INSERT](https://dev.mysql.com/doc/refman/8.0/en/insert.html) statement are ignored. For example, without IGNORE, a row that duplicates an existing UNIQUE index or PRIMARY KEY value in the table causes a duplicate-key error and the statement is aborted. With IGNORE, the row is discarded and no error occurs. Ignored errors generate warnings instead.` https://dev.mysql.com/doc/refman/8.0/en/insert.html suggests that an error will be thrown if PK already exists so need to use IGNORE or WHERE NOT EXISTS

Reply:
IMO "where not exists" approaches obfuscation, especially in the midst of both the prepare statement `?` placeholders and the `%s` formats for the table name.  given the baseline cost of a remote DB call, an exception is not terribly expensive.  thus, I'd recommend `INSERT` w/o `IGNORE` and then to be prepared to catch it

Reply:
Updated to catch the error